src/gridfire/simulations.clj

```diff
@@ -76,17 +81,14 @@
              (->
                (outputs/exec-in-outputs-writing-pool
                  (fn []
-                   (let [matrix (if (= name "burn_history")
-                                  (to-color-map-values layer output-time)
-                                  (fire-spread-results layer))]
-                     (layer-snapshot burn-time-matrix matrix output-time))))
-               (mfd/chain
```

Comment:
Explanation: this `mfd/chain` was at the heart of the issue. For half a dozed layers and hundreds of hourly timestepped, the `(layer-snapshot ...)` call would be executed roughly in parallel, returning intermediate rasters that all stuck around at the same time in memory while instances of the second callback function were queued `(fn [filtered-matrix] ...)` was queued to be executed in the outputs-writing thread pool.

The solution is to avoid re-queueing such intermediary results: create them and write them to disk in one feel swoop.